js/dialogcards.js

```diff
@@ -69,27 +89,117 @@ H5P.Dialogcards = (function ($, Audio, JoubelUI) {
       '<div class="h5p-dialogcards-title"><div class="h5p-dialogcards-title-inner">' + self.params.title + '</div></div>' +
       '<div class="h5p-dialogcards-description">' + self.params.description + '</div>'
       ));
+      var existsCardOrder = true;
+      if ($.isEmptyObject(this.cardOrder)) {
+        existsCardOrder = false;
+      }
+      // Create cardOrder and cardNumber buttons only on first instanciation for logged in user.
+      if (self.params.behaviour.randomCards == 'user' && !existsCardOrder) {
+        self.createOrder().appendTo(self.$inner);
+      } else {
+        self.attachContinue();
+      }
+  };
+
+  C.prototype.attachContinue = function ($container) {
+    var self = this;
+
+    // Remove potential user interaction elements from DOM.
+    var $el = '.h5p-dialogcards-number';
+    $( $el ).remove();
 
     if (self.params.behaviour.scaleTextNotCard) {
-      $container.addClass('h5p-text-scaling');
+      self.$inner.addClass('h5p-text-scaling');
     }
 
-    self.initCards(self.params.dialogs)
+    self.initCards(self.dialogs)
       .appendTo(self.$inner);
 
     self.createFooter()
       .appendTo(self.$inner);
 
+    $feedback = $('<div class="h5p-dialogcards h5p-feedback h5p-dialogcards-disabled">' + self.params.finished + '</div>').appendTo(self.$inner);
+
     self.updateNavigation();
 
-    self.on('reset', function () {
-      self.reset();
+    self.on('retry', function () {
+      self.retry();
     });
 
     self.on('resize', self.resize);
     self.trigger('resize');
   };
 
+
+  /**
+   * Create orderCards option request
+   *
+   * @returns {*|jQuery|HTMLElement} Order element
+   */
+  C.prototype.createOrder = function () {
+    var self = this;
+    var $order = $('<div>', {
+      'class': 'h5p-dialogcards-order',
+      'style': 'height:200px;',
+      'html': self.params.randomizeCardsQuestion + "&nbsp;&nbsp;",
+      'style': 'height:200px;'
+    });
+
+    self.$normalOrder = JoubelUI.createButton({
+      'class': 'h5p-dialogcards-order-button',
+      'title': self.params.no,
+      'html': self.params.no
+    }).click(function () {
+      self.randomizeOrder("normal");
+    }).appendTo($order);
+
+    self.$randomizeOrder = JoubelUI.createButton({
+      'class': 'h5p-dialogcards-order-button',
+      'title': self.params.yes,
+      'html': self.params.yes
+    }).click(function () {
+      self.randomizeOrder("random");
+    }).appendTo($order);
+
+    return $order;
+  };
+
+  C.prototype.createNumberCards = function () {
+    var self = this;
+
+    var $nbAllCards = self.params.dialogs.length;
+
+    var $numberCards = $('<div>', {
+      'class': 'h5p-dialogcards-number',
+      'style': 'height:200px;',
+      'html': self.params.nbCardsQuestion + "<br /><br />"
+    });
+
+    for (var i = 5; i < $nbAllCards; i+= 5) {
```

Comment:
Missing a comment here. When speedreading it is hard to understand why it is starting at 5 and incrementing with 5